text/0000-esc-escape.md

```diff
@@ -0,0 +1,38 @@
+- Feature Name: esc_escape
+- Start Date: 2015-12-29
+- RFC PR: (leave this empty)
+- Rust Issue: (leave this empty)
+
+# Summary
+[summary]: #summary
+
+Add a byte escape `\e` as shorthand for 0x1B (ESC), similar to GCC's `\e`.
+
+# Motivation
+[motivation]: #motivation
+
+ANSI escape codes are a vital part of CLIs. These are used both in stdin (special keys, control commands, etc.) and stdout (colors, text formatting, change in previous text, movement of the cursor, etc.). Adding a byte escape for it will make it easier to read the source code of and write interactive CLIs.
+
+Many compilers do already have this byte escape, GCC is an example (however, `\e` is not a part of ISO C).
+
+# Detailed design
+[design]: #detailed-design
+
+Add a byte escape `\e` as shorthand for 0x1B (ESC) to characters, strings, byte characters, and byte strings. Furthermore, add it to the escape iterators.
+
+# Drawbacks
+[drawbacks]: #drawbacks
+
+None.
```

Comment:
There is a definite drawback here; adding more escape codes increases the incidental complexity of the language (and [graydon cited some other common escapes as something he was glad Rust did not have](﻿﻿https://graydon2.dreamwidth.org/218040.html)).